File: system/extras/simpleperf/demo/SimpleperfExamplePureJava/app/src/main/java/com/example/simpleperf/simpleperfexamplepurejava/SleepActivity.java
package com.example.simpleperf.simpleperfexamplepurejava;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class SleepActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sleep);
        createRunSleepThread();
    }

    void createRunSleepThread() {
        new Thread(new Runnable() {
            volatile int counter = 0;
            long totalRunTimeInNs = 0;
            long totalSleepTimeInNs = 0;

            private long RunFunction() {
                long startTimeInNs = System.nanoTime();
                for (int i = 0; i < 10000000; ++i) {
                    counter = callFunction(counter);
                }
                return System.nanoTime() - startTimeInNs;
            }

            private long SleepFunction(long sleepTimeInNs) {
                long startTimeInNs = System.nanoTime();
                try {
                    Thread.sleep(sleepTimeInNs / 1000000, (int) (sleepTimeInNs % 1000000));
                } catch (Exception e) {
                }
                return System.nanoTime() - startTimeInNs;
            }

            @Override
            public void run() {
                while (true) {
                    totalRunTimeInNs += RunFunction();
                    if (totalSleepTimeInNs < totalRunTimeInNs) {
                        totalSleepTimeInNs += SleepFunction(
                                totalRunTimeInNs - totalSleepTimeInNs);
                    }
                }
            }

            private int callFunction(int a) {
                return a+1;
            }
        }, "RunSleepThread").start();
    }

}

File: system/extras/simpleperf/demo/SimpleperfExampleOfKotlin/app/src/main/java/com/example/simpleperf/simpleperfexampleofkotlin/SleepActivity.kt
package com.example.simpleperf.simpleperfexampleofkotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle

class SleepActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sleep)
        createRunSleepThread();
    }

    fun createRunSleepThread() {
        object : Thread() {
            var counter = 0
            var totalRunTimeInNs: Long = 0
            var totalSleepTimeInNs: Long = 0

            fun RunFunction(): Long {
                var startTimeInNs = System.nanoTime()
                var i = 0
                while (i < 10000000) {
                    counter = callFunction(counter)
                    i++
                }
                return System.nanoTime() - startTimeInNs
            }

            fun SleepFunction(sleepTimeInNs: Long): Long {
                var startTimeInNs = System.nanoTime()
                Thread.sleep(sleepTimeInNs / 1000000, (sleepTimeInNs % 1000000).toInt())
                return System.nanoTime() - startTimeInNs
            }

            override fun run() {
                while (true) {
                    totalRunTimeInNs += RunFunction()
                    if (totalSleepTimeInNs < totalRunTimeInNs) {
                        totalSleepTimeInNs += SleepFunction(totalRunTimeInNs - totalSleepTimeInNs)
                    }
                }
            }

            fun callFunction(i: Int): Int {
                return i + 1
            }
        }.start()
    }
}
